### ISSUE TITLE
Conflict between `Style/RedundantParentheses` and `Style/ParenthesesAroundCondition`

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop/rubocop#12537]
## Expected behavior

No offenses from `Style/RedundantParentheses` when I have `Style/ParenthesesAroundCondition` with `AllowInMultilineConditions: true`.

## Actual behavior

There is an offense:

![image](https://github.com/rubocop/rubocop/assets/6510020/03014b36-95d1-4374-aa96-a43664724e8e)

## Steps to reproduce the problem

```ruby
if (
  interaction.request.uri.include?('streams/key') &&
    interaction.response.headers['content-type']
      &.any? { |type| type.include?('application/json') }
)
  # ...
end
```

```yaml
Style/ParenthesesAroundCondition:
  AllowInMultilineConditions: true
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-performance 1.19.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.24.1
```
